xdsl/dialects/experimental/hls.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+
+from xdsl.dialects.builtin import (
+    Attribute,
+    StringAttr,
+)
+from xdsl.ir import Operation, SSAValue, Dialect
+from xdsl.irdl import (
+    irdl_op_definition,
+    Operand,
+    IRDLOperation,
+    operand_def,
+    opt_attr_def,
+)
+from xdsl.dialects.builtin import i32
+
+
+@irdl_op_definition
+class PragmaPipeline(IRDLOperation):
+    name = "hls.pipeline"
+    ii: Operand = operand_def(i32)
+
+    @staticmethod
+    def get(ii: SSAValue | Operation):
```

Comment:
I would recommend using `__init__`. We haven't migrated the whole project yet but it's the recommended way of creating operations these days. The newly added Linalg dialect might be a good example.

Reply:
Just changed that. Thank you :).

Also, I have added multiple other changes. Should I open a new pull request or can I ask for another review on this one?

Reply:
@gabrielrodcanal I think re-requesting review is the way to go and you can also comment inline in the code for things you want people to have a closer look at, if that's the case.

Reply:
@compor I honestly think it's ready to merge. Should I create a different pull request then, since this is a draft?

Reply:
No, you can just remove the draft status and delete the [WIP] from the title.